### ISSUE TITLE
Which packages to export a full_integer_quant INT8 tflite Yolov8 model?

### ISSUE THREAD

**Issue by fanchb** [ultralytics/ultralytics#20649]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

To be able to run an object detection model on an Android DSP, I need a full quantized INT8 tfilte version. I have tried to export my model with YoloV8 and different packages versions with no success (several different errors reported in past issues). I started from YoloV8.3.99 tag with following packages:

- Python 3.8
- tensorflow 2.13.0
- tf-keras 2.13.1
- onnx2tf 1.17.0 and later onnx2tf 1.20.0

Are there known package versions which work for that INT8 export with most recent YoloV8 tag? Are Python 3.10 and onnx2tf 1.26.3 required? 

Thank you for your help!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @fanchb, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started while an Ultralytics engineer reviews your question.

If this is a 🐛 Bug Report, please provide a minimum reproducible example (MRE) if you haven't already. This will help us understand and debug the issue more effectively.

If this is a custom training ❓ Question, please include as much detail as possible, such as dataset image examples, training logs, and confirmation that you're following our recommended training tips for best results.

We also encourage you to join the Ultralytics community for real-time discussions and support. You can connect with others on Discord, Discourse, or the Ultralytics Subreddit to share your experience and get help from fellow users.

## Upgrade

Please ensure you are using the latest version of the `ultralytics` package and all requirements in a Python 3.8 or later environment with PyTorch 1.8 or newer to verify your issue is not already addressed in the most recent release. Upgrade using:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in a variety of up-to-date verified environments, including cloud notebooks with free GPU, Google Cloud Deep Learning VM, Amazon Deep Learning AMI, and Docker images. Make sure your environment includes all necessary dependencies such as CUDA, CUDNN, Python, and PyTorch.

## Status

Ultralytics CI tests run every 24 hours and on every commit to verify correct operation across all YOLO modes and tasks on macOS, Windows, and Ubuntu.

An Ultralytics engineer will review your issue and assist you soon. Thank you for your patience and for being part of the Ultralytics community!

**Comment by Y-T-G** [comment]
Why not use the latest Ultralytics?

**Comment by fanchb** [comment]
@Y-T-G , good point !

I wanted to compare between FP16 (running on a GPU but overheating at high inference rate) and INT8 (running on the DSP) performances and I believed I needed to stay on YoloV8 since there was an issue with GPU delegate in YoloV11. But you just fixed it with commits described in #20436. Wonderful!

So, with latest Ultralytics, no known limitations in the export of a best_full_interger_quant model?

Thank you!

**Comment by Y-T-G** [comment]
YOLOv8 works with any Ultralytics version. Ultralytics is not limited to single model.

**Comment by fanchb** [comment]
Hi,

After your suggestions, I trained two models, one based on V8 and the other one on V11 with code like:

# Load a model
model = YOLO("yolov8s.pt")  # load a pretrained model (recommended for training)

or 

model = YOLO("yolo11s.pt") 

# Train using the single most idle GPU
results = model.train(...)

I have noticed that both resulting models have different number of parameters (11177234 versus 9464506) which is good. But their evaluation on my test dataset gives almost the same Recall and Precision performances.

On top of that, if I request the 50 first operators, they are exactly strictly the same for both models. 

1. PAD
2. CONV_2D
3. LOGISTIC
4. MUL
5. MUL
6. MUL
7. MUL
8. MUL
9. PAD
10. CONV_2D
11. LOGISTIC
12. MUL
13. MUL
14. MUL
15. MUL
16. MUL
17. CONV_2D
18. LOGISTIC
19. MUL
20. MUL
21. MUL
22. MUL
23. MUL
24. STRIDED_SLICE
25. STRIDED_SLICE
26. CONV_2D
27. LOGISTIC
28. MUL
29. MUL
30. MUL
31. MUL
32. MUL
33. CONV_2D
34. LOGISTIC
35. MUL
36. MUL
37. MUL
38. MUL
39. MUL
40. ADD
41. CONCATENATION
42. CONV_2D
43. LOGISTIC
44. MUL
45. MUL
46. MUL
47. MUL
48. MUL
49. PAD
50. CONV_2D

It looks more related to YoloV8 architecture than YoloV11 with its (CONV_2D → LOGISTIC → MUL) repeated pattern.

 Could it be that I did train YoloV8 models in both cases? 

Thank you!

**Comment by glenn-jocher** [comment]
@fanchb The different parameter counts (11,177,234 vs 9,464,506) confirm you're using different model architectures - YOLO11 is indeed more efficient with fewer parameters. 

The identical TFLite operator sequences you're seeing are likely due to the export conversion process through ONNX and onnx2tf, which can normalize certain operations. To verify you're actually training different models, check the model summary before export:

```python
model = YOLO("yolo11s.pt")
print(model.model)  # This will show the actual architecture
```

The architectural differences between YOLOv8 and YOLO11 should be visible in the PyTorch model structure, but may appear similar after TFLite conversion due to operator fusion and optimization during the export process.
